roadmap/implementers-guide/src/runtime/disputes.md

```diff
@@ -17,90 +19,351 @@ We account for these requirements by having the validity module handle two kinds
+The first is a remote roll-up of a **concluded dispute**. These are simply all attestations for the block, those against it, and the result of all (secondary) approval checks. A concluded remote dispute can be resolved in a single transaction as it is an open-and-shut case of a quorum of validators disagreeing with another.
 
-## Remote Disputes
+The second type of remote dispute is the **unconcluded dispute**. An unconcluded remote dispute is started by any validator using a `UnconcludedDisputeGossipNotificationMessage`:
 
-When a dispute has occurred on another fork, we need to transplant that dispute to every other fork. This poses some major challenges.
+When beginning a remote dispute, at least one escalation by a validator is required, but this validator may be malicious and desires to be slashed. There is no guarantee that the para is registered on this fork of the relay chain or that the para was considered available on any fork of the relay chain.
 
-There are two types of remote disputes. The first is a remote roll-up of a concluded dispute. These are simply all attestations for the block, those against it, and the result of all (secondary) approval checks. A concluded remote dispute can be resolved in a single transaction as it is an open-and-shut case of a quorum of validators disagreeing with another.
+The first step is the escalation, to notify peers of the unconcluded gossip.
 
-The second type of remote dispute is the unconcluded dispute. An unconcluded remote dispute is started by any validator, using these things:
+## Disputing an Unavailable Candidate
 
-- A candidate
-- The session that the candidate has appeared in.
-- Backing for that candidate
-- The validation code necessary for validation of the candidate.
-  > TODO: optimize by excluding in case where code appears in `Paras::CurrentCode` of this fork of relay-chain
-- Secondary checks already done on that candidate, containing one or more disputes by validators. None of the disputes are required to have appeared on other chains.
-  > TODO: validator-dispute could be instead replaced by a fisherman w/ bond
+In the case of malicious validator sending a dispute gossip message, with a block hash, that never became part of any parachain.
+
+If the candidate was not made available on another fork of the relay chain (as in it does not exist), the availability process will time out and the disputing validator will be slashed on this fork.
+As with other disputes, the escalation
+
+> TODO: is the escalation equivalent to the `DisputeGossipMessage` or is it something separate? Which information must be contained in that.
+
+## Resolving state of availability
+
+The availbility or unavailability of a block disputed in an escalation / `DisputeGossipMessage`.
```

Comment:
It'll be simplest if a dispute always brings an availability run, but just independently without associating it to any core.  We then continue this availability run and then begin reconstruction.,  I suppose reconstruction could begin even before availability

We face a choice here: Do we check availability on-chain or not?
- If off-chain, then we require another "not available" vote if we want our on-chain to understand that availability failed, but not sure if we even care.  
- If on-chain, then we face another choice:
  - We could simply replay availability votes here, which sounds okay.  
  - We could avoid replay by doing an availability core roll up like this: 
     - Inside `DisputeGossipMessage`, we add a single Merkle proof from some relay chain block hash, not necessarily one in the current chain.  If you cannot do this proof into the block then you could maybe do two proofs into the state before and after or something.  We then know the earliest relay chain block referenced by availability votes, so we can prove that availability votes referred to the right core maybe.  Ugh!

I'd say keep availability off-chain and we'll discuss with Al and/or Rob if we need it on-chain for any reason.  I suppose this still runs into issues if we never imported some relay chain blocks, so maybe this still requires replay or something.